src/confluent_kafka/avro/__init__.py

```diff
@@ -54,8 +54,8 @@ def __init__(self, config, default_key_schema=None,
             sr_conf['sasl.mechanism'] = config.get('sasl.mechanism', config.get('sasl.mechanisms', ''))
             sr_conf['sasl.username'] = config.get('sasl.username', '')
             sr_conf['sasl.password'] = config.get('sasl.password', '')
-            sr_conf['auto.register.schemas'] = config.get('auto.register.schemas', True)
 
+        sr_conf['auto.register.schemas'] = config.get('auto.register.schemas', True)
```

Comment:
Does this really fix the problem?
I think `auto.register.schemas` must also be removed from `config`.

Reply:
the problem is that auto.register.schemas is not well indented  ( it's only take in account if basic.auth.credentials.source is equal to SASL_INHERIT , and it should not be related )

why do you want auto.register.schemas fro mthe config ? ( it's a very useful option )